toolchain/defs.bzl

```
@@ -78,29 +80,44 @@ def toolchains(
         url_formats = [mirror_format, original_format]
 
     zig_repository(
-        name = "zig_sdk",
+        name = "zig_sdk-{}".format(exec),
         version = version,
         url_formats = url_formats,
         host_platform_sha256 = host_platform_sha256,
         host_platform_ext = host_platform_ext,
+        exec = exec,
     )
 
 def _quote(s):
     return "'" + s.replace("'", "'\\''") + "'"
 
 def _zig_repository_impl(repository_ctx):
-    arch = repository_ctx.os.arch
-    if arch == "amd64":
-        arch = "x86_64"
-
-    os = repository_ctx.os.name.lower()
-    if os.startswith("mac os"):
-        os = "macos"
-
-    if os.startswith("windows"):
-        os = "windows"
-
-    host_platform = "{}-{}".format(os, arch)
+    arch_os = repository_ctx.attr.exec.split("-")
+    exec_os = arch_os[0]
+    exec_arch = arch_os[1]
```

Let's pass the exec_os and exec_arch as attributes, so we don't need to split here.
```suggestion
    exec_os = repository_ctx.attr.exec_os
    exec_arch = repository_ctx.attr.exec_arch
```